tools/clang/lib/Sema/SemaStmt.cpp

```diff
@@ -2709,6 +2701,37 @@ VarDecl *Sema::getCopyElisionCandidate(QualType ReturnType,
   if (!VD)
     return nullptr;
 
+  // HLSL Change Begins: NRVO unsafe for a variety of cases in HLSL
+  // - vectors/matrix with bool component types
+  // - attributes not captured to QualType, such as precise and globallycoherent
+  // NRVO is also unnecessary for HLSL correctness or performance.
+  if (getLangOpts().HLSL) {
+    // exclude vectors/matrix (not treated as record type)
+    // NRVO breaks on bool component type due to diff between
+    // i32 memory and i1 register representation
+    if (hlsl::IsHLSLVecMatType(ReturnType))
+      return false;
```

Comment:
Returning `false` from a function that returns a pointer is not ideal, these should be `nullptr`.